test/common/config/config_provider_impl_test.cc

```diff
@@ -195,19 +195,11 @@ class DummyConfigProviderManager : public ConfigProviderManagerImplBase {
 
   // Envoy::Config::ConfigProviderManager
   ConfigProviderPtr
-  createStaticConfigProvider(const Protobuf::Message& config_proto,
+  createStaticConfigProvider(ProtobufTypes::ConstMessagePtrVector&& configs,
                              Server::Configuration::ServerFactoryContext& factory_context,
-                             const Envoy::Config::ConfigProviderManager::OptionalArg&) override {
-    return std::make_unique<StaticDummyConfigProvider>(
-        dynamic_cast<const test::common::config::DummyConfig&>(config_proto), factory_context,
-        *this);
-  }
-  ConfigProviderPtr
-  createStaticConfigProvider(std::vector<std::unique_ptr<const Protobuf::Message>>&&,
-                             Server::Configuration::ServerFactoryContext&,
                              const OptionalArg&) override {
-    ASSERT(false, "this provider does not expect multiple config protos");
-    return nullptr;
+    auto config = dynamic_cast<const test::common::config::DummyConfig&>(*configs[0]);
```

Comment:
nit: add `ASSERT(configs.size() == 1);` and a comment saying that currently the provide does not expect multiple config protos.

Reply:
https://github.com/envoyproxy/envoy/pull/34985/commits/3f86042de1e00152c52ea84b0fa2eaaf9d021a51 👍